tools/release-management/src/bin/version_updater.rs

```diff
@@ -0,0 +1,208 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use git_cmd::Repo;
+use time::format_description;
+use toml_edit::DocumentMut;
+
+const COMMIT_MESSAGE: &str = "chore: Update Rust & Go SDK versions";
+const PR_TITLE: &str = "Update SDK versions";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(short, long, default_value = "../..")]
+    root: String,
+
+    #[arg(short, long, default_value = ".")]
+    git_dir: String,
+
+    #[arg(
+        short = 'o',
+        long,
+        default_value = "coralogix/coralogix-management-sdk"
+    )]
+    github_repo: String,
+
+    #[arg(short, long, short = 'v', default_value = "0.2.1")]
+    sdk_version: String,
+
+    #[arg(short = 'c', long, default_value = "rust")]
+    rust_root: String,
+
+    #[arg(short = 'g', long, default_value = "go")]
+    go_root: String,
+}
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let args = Args::parse();
+    let cargo_toml_path = PathBuf::from(&args.root).join("rust").join("Cargo.toml");
+    let go_constants_path = PathBuf::from(&args.root).join("go").join("constants.go");
+    set_rust_sdk_version(&args, &cargo_toml_path).await?;
+    set_go_sdk_version(&args, &go_constants_path).await?;
+
+    let (owner, repo_name) = args
+        .github_repo
+        .split_once('/')
+        .expect("Invalid value for github_repo (use owner/repo format)");
+
+    let github_token = std::env::var("GITHUB_TOKEN").expect("A GITHUB_TOKEN must be set");
+
+    let repo = Repo::new(&args.git_dir)
+        .unwrap_or_else(|_| panic!("Couldn't find git repository at {}", args.git_dir));
+    let format = format_description::parse("[year]-[month]-[day]").expect("Failed to parse format");
+    let branch_name = format!(
+        "update-{}",
+        time::OffsetDateTime::now_utc().format(&format)?
+    );
+    let changes = repo.changes(|_| true).expect("Failed to add changes");
+    repo.checkout_new_branch(&branch_name)
+        .expect("Failed to create branch");
+    tracing::info!(%branch_name, ?changes);
+
+    repo.add(&[
+        go_constants_path.to_str().unwrap(),
+        cargo_toml_path.to_str().unwrap(),
+    ])
+    .expect("Failed to add files");
+    repo.commit(COMMIT_MESSAGE).expect("Failed to commit");
+    repo.push(&branch_name).expect("Failed to push");
+
+    // Create PR
+    let authed_github = octocrab::OctocrabBuilder::default()
+        .personal_token(github_token)
+        .build()?;
+
+    // tracing::info!(%owner, %repo_name, ?branch_name, ?pr_message);
+    authed_github
+        .pulls(owner, repo_name)
+        .create(PR_TITLE, branch_name, "master")
+        .body(COMMIT_MESSAGE)
+        .send()
+        .await?;
+
+    Ok(())
+}
+
+async fn set_rust_sdk_version(args: &Args, cargo_toml_path: &PathBuf) -> eyre::Result<()> {
```

Comment:
use a Read implementation instead of a pathbuf. this way you can pass in a string or a reader or anything else

https://users.rust-lang.org/t/how-to-use-a-std-read-as-a-function-parameter/97379

Reply:
Use tokio's Async read tho

Reply:
This no longer reads from fs, so now it just takes a string as input